samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java

```diff
@@ -2952,7 +2952,11 @@ public String toScientificNotation(String numberStr,int decimalPlaces){
 	    && bdx.doubleValue() != 0) {
 		  formatter = new DecimalFormat(FORMAT_MASK);
 	  } else {
-		  formatter = new DecimalFormat("0");
+		  if ((String.valueOf(bdx.doubleValue()).contains("e")) || ((String.valueOf(bdx.doubleValue()).contains("E")))) {
```

Comment:
Quite minor, but maybe we could use StringUtils.containsIgnoreCase or StringUtils.containsAny